File: array/problem26/problem26.cpp
#include <vector>

using namespace std;

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        // Write position k; first element is always kept when n >= 1
        int n = static_cast<int>(nums.size());
        if (n == 0) {
            return 0;
        }
        int k = 1;
        for (int i = 1; i < n; ++i) {
            if (nums[i] != nums[k - 1]) {
                nums[k] = nums[i];
                k++;
            }
        }
        return k;
    }
};

File: array/problem26/problem26.py
from typing import List


class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        # k is the length of the unique prefix; nums[k-1] is the last kept value
        if not nums:
            return 0
        k = 1
        for i in range(1, len(nums)):
            if nums[i] != nums[k - 1]:
                nums[k] = nums[i]
                k += 1
        return k
